internal/sirius/my_details_test.go

```diff
@@ -0,0 +1,89 @@
+package sirius
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type request struct {
+	method, path string
+	cookies      []*http.Cookie
+	headers      http.Header
+}
+
+func TestMyDetails(t *testing.T) {
+	assert := assert.New(t)
+	requests := make(chan request, 1)
+
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		requests <- request{
+			method:  r.Method,
+			path:    r.URL.Path,
+			cookies: r.Cookies(),
+			headers: r.Header,
+		}
+
+		_, err := io.WriteString(w, `{"id":47,"name":"system","phoneNumber":"03004560300","teams":[{"id":10,"name":"Allocations - (Supervision)","phoneNumber":"0123456789","teams":[],"displayName":"Allocations - (Supervision)","deleted":false,"tasks":[],"email":"allocations.team@opgtest.com"}],"displayName":"system admin","deleted":false,"tasks":[],"email":"system.admin@opgtest.com","firstname":"system","surname":"admin","roles":["OPG User","System Admin"],"locked":false,"suspended":false}`)
```

Comment:
For the future: it would be good to manage this mock more cleanly, whether that's pulling it from a JSON file or using something like Pact.

We'd then be able to clearly demonstrate what we expect from the API, and easily be able to downstream any changes. Otherwise it could be like the LPA frontend where the tests pass, but the mocked API response is nothing like the actual API.

Reply:
Pact is a good idea, I had totally forgotten about that as something to look at